src/cuda/cuda_fft.cu

```diff
@@ -67,50 +67,24 @@ namespace
 namespace FFT_Cuda
 {
     ComplexData::ComplexData()
-        : _data  ( NULL )
-        , _width ( 0 )
-        , _height( 0 )
     {
     }
 
     ComplexData::ComplexData( const PenguinV_Image::Image & image )
-        : _data  ( NULL )
-        , _width ( 0 )
-        , _height( 0 )
     {
         set( image );
     }
 
     ComplexData::ComplexData( const ComplexData & data )
-        : _data  ( NULL )
-        , _width ( 0 )
-        , _height( 0 )
     {
         _copy( data );
     }
 
     ComplexData::ComplexData( ComplexData && data )
-        : _data  ( NULL )
-        , _width ( 0 )
-        , _height( 0 )
     {
         _swap( data );
     }
 
-    ComplexData & ComplexData::operator=( const ComplexData & data )
```

Comment:
How about `move` operator `ComplexData & ComplexData::operator=( ComplexData && data )` ?
Try this:
```cpp

ComplexData A = f();

where:

ComplexData f()
{
    Image image(100, 100);
    ComplexData a(image);
    return a;
}
```

We shouldn't call a copy function in this case.

Reply:
The `move` version of `ComplexData & ComplexData::operator=( ComplexData && data)` is also inside the `BaseComplexData` class template, but it uses the member function `_swap()` which isn't abstract. The template parameter gives the class template `BaseComplexData` enough information to do the swaps as they were originally implemented.

Reply:
@ihhub I'm guessing everything is okay with the setup of `operator=()` since you haven't commented on my explanation. So I'm going to mark the conversation on `operator()=` as resovled.